egs2/slue-voxpopuli/slu1/local/prepare_nel_data.py

```diff
@@ -0,0 +1,169 @@
+"""
+Extract entity phrase and duration tuples for evaluation
+"""
+
+import json
+import os
+import sys
+
+from datasets import load_dataset
+from fire import Fire
+from tqdm import tqdm
+
+
+def save_json(fname, dict_name):
+    with open(fname, "w") as f:
+        f.write(json.dumps(dict_name, indent=4))
+
+
+def depunctuate(text):
+    """
+    Also removes isolated apostrophe
+    """
+    text = text.replace(".", "")
+    text = text.replace("' ", " ")
+    text = text.replace("'s", " 's")
+    text = text.replace("  ", " ")
+    return text
+
+
+def read_gt_sample(gt_labels, data_idx):
+    entity_phrase = depunctuate(gt_labels[data_idx][0])
+    wrd_lst = entity_phrase.split(" ")
+    return wrd_lst, len(wrd_lst)
+
+
+def update_alignment_dct(all_word_alignments, entity_alignments, utt_id, gt_labels):
+    """
+    Update alignment_dct by appending word belonging to entity phrases with #
+    """
+    text = " ".join([item[0] for item in all_word_alignments[utt_id]])
+    text = text.replace("   ", " ")
+    text = text.replace("  ", " ")
+    if len(gt_labels) > 0:
+        data_idx, curr_idx = 0, 0
+        wrd_lst, len_phrase = read_gt_sample(gt_labels, data_idx)
+        while data_idx < len(gt_labels):  # until a match for all GT entities is found
+            label, _, _ = all_word_alignments[utt_id][curr_idx]
+            if label == wrd_lst[0]:
+                if label == "committee":
+                    import pdb
+
+                    pdb.set_trace()
+                update_words = True
+                if len_phrase > 1:
+                    done_processing = False
+                else:
+                    done_processing = True
+                gt_idx = 1
+                tier_idx = 1
+                while not done_processing:
+                    label, _, _ = all_word_alignments[utt_id][curr_idx + tier_idx]
+                    if label != "":
+                        if (
```

Comment:
Simplify the logic here as now we are handling apostrophe in `modify_word_alignments`

```
                    if label != "":
                        if label != wrd_lst[gt_idx]:
                            done_processing = True
                            update_words = False
                        else:
                            gt_idx += 1
```

Reply:
Fixed